ops/pebble.py

```diff
@@ -2172,29 +2485,44 @@ def get_checks(
         Returns:
             List of :class:`CheckInfo` objects.
         """
-        query = {}
-        if level is not None:
+        query = {}  # type: Dict[str, Union[str, List[str]]]
+        if level:
             query['level'] = level.value
         if names:
-            query['names'] = names
-        resp = self._request('GET', '/v1/checks', query)
-        return [CheckInfo.from_dict(info) for info in resp['result']]
+            query['names'] = list(names)
+        resp = self._request('GET', '/v1/checks', query)  # type: ignore
+        info_dicts = typing.cast(List['_InfoDict'], resp['result'])
+        return [CheckInfo.from_dict(info) for info in info_dicts]
 
 
 class _FilesParser:
     """A limited purpose multi-part parser backed by files for memory efficiency."""
-
-    def __init__(self, boundary: typing.Union[bytes, str]):
-        self._response = None
-        self._files = {}
+    if TYPE_CHECKING:
+        # to help the type checker and IDEs:
+        @property
+        def _response(self) -> '_FilesResponse': ...  # noqa
+        @_response.setter
+        def _response(self, value: '_FilesResponse'): ...  # noqa
+        @property
+        def _headers(self) -> Optional['Message']: ...  # noqa
+        @_headers.setter
+        def _headers(self, val: Optional['Message']) -> None: ...  # noqa
+        @property
+        def _part_type(self) -> _PartType: ...  # noqa
+        @_part_type.setter
+        def _part_type(self, val: _PartType) -> None: ...  # noqa
```

Comment:
You're right, it should not obfuscate the code. So far my feeling is that the code gained in readability by adding the types.
I simplified those `@property` things away and gave the attributes direct `Optional[x]` types, that works too. I can't recall why I used that strategy, but it seems not to be necessary (any longer).
Do you feel it's still obfuscating instead of clarifying?